Modules/posixmodule.c

```diff
@@ -3869,7 +3880,18 @@ posix_fork(PyObject *self, PyObject *noargs)
     pid_t pid;
     int result = 0;
     _PyImport_AcquireLock();
+#ifdef WITH_THREAD
+    if (_PyThread_AcquireKeyLock() == 0) {
+        _PyImport_ReleaseLock();
+        PyErr_SetString(PyExc_RuntimeError,
+                        "could not acquire thread key lock");
+        return NULL;
+    }
+#endif
     pid = fork();
+#ifdef WITH_THREAD
+    _PyThread_ReleaseKeyLock();
```

Comment:
At least in the child process, I don't think that it's safe to use keymutex just after fork.

A few lines below,  PyOS_AfterFork() is called if we are in the child process, this function calls PyThread_ReInitTLS():

```
void
PyThread_ReInitTLS(void)
{
    ...
    /* As with interpreter_lock in PyEval_ReInitThreads()
       we just create a new lock without freeing the old one */
    keymutex = PyThread_allocate_lock();
    ...
}
```

So keymutex is replaced in the child process.

Reply:
Maybe a compromise would be only unlock in the parent process. _PyThread_ReleaseKeyLock(); in the parent is safe, whereas _PyThread_ReleaseKeyLock(); in the child can be simply... ignored, since we replace keymutex a few instructions later (see my previous comment).